src/socket.cpp

```diff
@@ -1685,7 +1685,11 @@ void SIPpSocket::sipp_sctp_peer_params()
         memset(&peerparam, 0, sizeof(peerparam));
 
         sockaddr* addresses;
+#ifdef __SUNOS
+        int addresscount = sctp_getpaddrs(ss_fd, 0, (void **) &addresses);
+#else
         int addresscount = sctp_getpaddrs(ss_fd, 0, &addresses);
```

Comment:
Solaris uses a different prototype unfortunately:
int sctp_getpaddrs(int sock, sctp_assoc_t id, void **addrs)

Without the cast to the cast to ```(void **)``` g++ throws an error:
```error: invalid conversion from ‘sockaddr**’ to ‘void**’```